gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,537 @@
+
+    private static final long serialVersionUID = -2830645032124504717L;
+
+    ZoneListOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the zones's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all zone fields are returned. {@code ZoneOption.fields} can
+     * be used to specify only the fields of interest. Zone ID is always returned, even if not
+     * specified. {@link ZoneField} provides a list of fields that can be used.
+     */
+    public static ZoneListOption fields(ZoneField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("nextPageToken,managedZones(").append(ZoneField.selector(fields)).append(')');
+      return new ZoneListOption(DnsRpc.Option.FIELDS, builder.toString());
+    }
+
+    /**
+     * Returns an option to specify a page token.
+     *
+     * <p>The page token (returned from a previous call to list) indicates from where listing should
+     * continue.
+     */
+    public static ZoneListOption pageToken(String pageToken) {
+      return new ZoneListOption(DnsRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Restricts the list to only zone with this fully qualified domain name.
+     */
+    public static ZoneListOption dnsName(String dnsName) {
+      StringBuilder builder = new StringBuilder();
+      return new ZoneListOption(DnsRpc.Option.DNS_NAME, dnsName);
+    }
+
+    /**
+     * The maximum number of zones to return per RPC.
+     *
+     * <p>The server can return fewer zones than requested. When there are more results than the
+     * page size, the server will return a page token that can be used to fetch other results.
+     */
+    public static ZoneListOption pageSize(int pageSize) {
+      return new ZoneListOption(DnsRpc.Option.PAGE_SIZE, pageSize);
+    }
+  }
+
+  /**
+   * Class for specifying project options.
+   */
+  class ProjectOption extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 6817937338218847748L;
+
+    ProjectOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the project's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all project fields are returned. {@code
+     * ProjectOption.fields} can be used to specify only the fields of interest. Project ID is
+     * always returned, even if not specified. {@link ProjectField} provides a list of fields that
+     * can be used.
+     */
+    public static ProjectOption fields(ProjectField... fields) {
+      return new ProjectOption(DnsRpc.Option.FIELDS, ProjectField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying change request field options.
+   */
+  class ChangeRequestOption extends AbstractOption implements Serializable {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://www.codacy.com/app/mziccard/gcloud-java/file/2317493773/issues/source?bid=3123109&fileBranchId=3203698#l347)